src/lld/backend_patterns/repository/template/Correction.md

```diff
@@ -4,6 +4,76 @@
 # DTO question
 - Q: What is the best practises for DTO objects for 2 or more entity relation?
 - A: DTO can have builder pattern to build required inputs if it seems to more combinations
+- Q: Is Deserialization of json to builder possible?
+- A: Yes, no arg constructor and static builder class is a way for parser.
+- Q: Example
+A: 
+```
+>> invoking code on deserialization
+ObjectMapper mapper = new ObjectMapper();
+Person person = mapper.readValue(jsonString, Person.Builder.class).build();
+
+>> json
+{
+  "name": "John",
+  "age": 30,
+  "address": {
+    "street": "123 Main St",
+    "city": "Anytown",
+    "state": "CA",
+    "zip": "12345"
+  }
```

Comment:
What happens in case of Exceptions?
What about nested json?

Reply:
got it. Please add these points as comments to the document

Reply:
Sure~ Also, wanted to know how your code base is handling those compute intensive scenarios?
we can discuss that

Reply:
DI for most cases. single objectMapper entity and use it across application